src/bin/lint.ts

```diff
@@ -1,39 +1,45 @@
 #!/usr/bin/env node
+import type { CLIOptions } from '../types.js';
 import os from 'node:os';
 import path from 'node:path';
 import process from 'node:process';
 import childProcess from 'node:child_process';
 import fs from 'node:fs';
+import { Command } from 'commander';
 import * as utils from '../utils.js';
 
 const platform = os.platform();
+const program = new Command();
+const DEFAULT_SHELLCHECK_SEARCH_ROOTS = ['./src', './scripts', './tests'];
+
+program
+  .name('matrixai-lint')
+  .description(
+    'Lint source files, scripts, and markdown with configured rules.',
+  )
+  .option('-f, --fix', 'Automatically fix problems')
+  .option(
+    '--user-config',
+    'Use user-provided ESLint config instead of built-in one',
+  )
+  .option('--config <path>', 'Path to explicit ESLint config file')
+  .option('--eslint <pat...>', 'Glob(s) to pass to ESLint')
+  .option('--shell  <pat...>', 'Glob(s) to pass to shell-check')
+  .allowUnknownOption(true); // Optional: force rejection of unknown flags
 
 /* eslint-disable no-console */
 async function main(argv = process.argv) {
-  argv = argv.slice(2);
+  await program.parseAsync(argv);
+  const options = program.opts<CLIOptions>();
+
```

Comment:
This isn't how we use Commander. The main code needs to be in the `.action(async () => {})` callback of the program. https://www.npmjs.com/package/commander#action-handler